scripts/missing-ingest-data-fixer/ingest_query.py

```diff
@@ -0,0 +1,180 @@
+                        exists = True
+                        in_ingest[project_ingest.get('uuid').get('uuid')] = project_dts
+                        break
+                if exists:
+                    exists = False
+                    break
+
+        return in_ingest
+
+    @staticmethod
+    def not_exists_by_geo_accession(ingest_projects, dts_projects):
+        exists = False
+        not_in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                for geo in project_ingest.get('content', {}).get('geo_series_accessions', []):
+                    if geo in project_dts['data_accession']:
+                        exists = True
+                        break
+                if exists:
+                    exists = False
+                    break
+            else:
+                not_in_ingest[project_dts['dcp_id']] = project_dts
+
+        return not_in_ingest
+
+    @staticmethod
+    def exists_by_geo_accession(ingest_projects, dts_projects):
+        exists = False
+        in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                for geo in project_ingest.get('content', {}).get('geo_series_accessions', []):
+                    if geo in project_dts['data_accession']:
+                        exists = True
+                        in_ingest[project_ingest.get('uuid').get('uuid')] = project_dts
+                        break
+                if exists:
+                    exists = False
+                    break
+
+        return in_ingest
+
+    @staticmethod
+    def filter_by_is_in_catalogue(ingest_projects, dts_projects):
+        not_in_cat = []
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                if 'isInCatalogue' not in project_ingest or project_ingest.get('isInCatalogue') == False:
+                    not_in_cat.append(project_dts)
+                    break
+
+        return not_in_cat
+
+    @staticmethod
+    def get_projects_without_organ(ingest_projects):
+        project_with_missing_organ = {}
+        for project in ingest_projects:
+            if 'organ' not in project or project.get('organ') is None:
+                project_with_missing_organ[project.get('uuid').get('uuid')] = project
+            elif 'ontologies' not in project.get('organ') or project.get('organ').get('ontologies') is None:
+                project_with_missing_organ[project.get('uuid').get('uuid')] = project
+            elif len(project.get('organ').get('ontologies')) == 0:
+                project_with_missing_organ[project.get('uuid').get('uuid')] = project
+
+        return project_with_missing_organ
+
+    @staticmethod
+    def get_projects_without_technology(ingest_projects):
+        project_with_missing_technology = {}
+        for project in ingest_projects:
+            if 'technology' not in project or project.get('technology') is None:
```

Comment:
The check for key is redundant, project.get('technology') should be enough

Reply:
No, it is not, because both of them could be true. `technology` might be missing from the payload or it could be there, but it could be `None`.

Reply:
project.get('technology') would return None if technology is not in project